source/client-side-operations-timeout/tests/client-side-encryption.yml

```diff
@@ -0,0 +1,79 @@
+
+tests:
+  - description: "timeoutMS applied to listCollections to get collection schema"
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 1 }
+      data:
+        failCommands: ["listCollections"]
+        blockConnection: true
+        blockTimeMS: 15
+    clientOptions:
+      autoEncryptOpts:
+        kmsProviders:
+          aws: {} # Credentials filled in from environment.
+      timeoutMS: 10
+    operations:
+      - name: insertOne
+        arguments:
+          document: &doc0 { _id: 1, encrypted_string: "string0", random: "abc" }
+        result:
+          timeoutError: true
+    expectations:
+      # Auto encryption will request the collection info.
+      - command_started_event:
+          command:
+            listCollections: 1
+            filter:
+              name: *collection_name
+          command_name: listCollections
+  - description: "timeoutMS applied to find to get keyvault data"
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 1 }
+      data:
+        failCommands: ["find"]
+        blockConnection: true
+        blockTimeMS: 15
+    clientOptions:
+      autoEncryptOpts:
+        kmsProviders:
+          aws: {} # Credentials filled in from environment.
+        timeoutMS: 10
+    operations:
+      - name: insertOne
+        arguments:
+          document: *doc0
+        result:
+          timeoutError: true
+    expectations:
+      # Auto encryption will request the collection info.
+      - command_started_event:
+          command:
+            listCollections: 1
+            filter:
+              name: *collection_name
+          command_name: listCollections
+      - command_started_event:
+          command:
+            listCollections: 1
+            filter:
+              name: "datakeys"
+            $db: keyvault
+          command_name: listCollections
+      # Then key is fetched from the key vault.
+      - command_started_event:
+          command:
+            find: datakeys
+            filter: {"$or": [{"_id": {"$in": [ {'$binary': {'base64': 'AAAAAAAAAAAAAAAAAAAAAA==', 'subType': '04'}} ] }}, {"keyAltNames": {"$in": []}}]}
+            $db: keyvault
+            readConcern: { level: "majority" }
```

Comment:
[listCollections](https://docs.mongodb.com/manual/reference/command/listCollections/) does not support maxTimeMS from the looks of it. But I think this `find` should include `maxTimeMS`, right?

Reply:
From talking to the server team, my understanding was that all commands support `maxTimeMS` as it's a top-level field that's parsed before the server gets to command-specific logic. I updated the tests to include it in command monitoring expectations. This also required a small update to the FLE test format to support the array-based `$$type`` syntax like the unified test format does.

Reply:
Ah I see. Thanks for adding it.